### PR TITLE
[4.0] - nested set rebuild/fix  for #__menu table

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes
apparently we don't have any issues, but if you look more deep maybe you'll discover that the nested set  values (parent_id, lft, rgt)  for the `#__menu` table are incorrect, or at least i believe so, silly me

### Testing Instructions
the most easy way i've found  need that you run this procedure on your test environment
i've made a gist for you https://gist.github.com/alikon/fc3bdaadcd999a29c9b2b97c00e3214e
just change `my4` db prefix with yours 
and then compare what is in your #__menu table with tmp_tree table
(e.g)

``` SELECT `id` , `parent_id`, `lft`, `rgt` FROM `my4_menu` ORDER by `parent_id` ```

with 

``` select * from `tmp_tree` ORDER by `parent_id` ```

### Expected result
no difference in `(id, parent_id, lft, rgt) `

### Actual result
quite some difference 

### Additional comment
if and when someone will confirm this the same should be done on the postgresql side which actually have even different values for (lft,rgt) for the same item id ....even without this pr...

### CODE REVIEW

File: installation/sql/mysql/joomla.sql [hunk 1]
```diff
@@ -1319,45 +1319,26 @@ CREATE TABLE IF NOT EXISTS `#__menu` (
...
+(101, 'mainmenu', 'Home', 'home', '', 'home', 'index.php?option=com_content&view=featured', 'component', 1, 1, 1, 20, 0, '0000-00-00 00:00:00', 0, 1, '', 0, '{featured_categories:[],layout_type:blog,num_leading_articles:1,num_intro_articles:3,num_columns:3,num_links:0,multi_column_order:1,orderby_pri:,orderby_sec:front,order_date:,show_pagination:2,show_pagination_results:1,show_title:,link_titles:,show_intro:,info_block_position:,show_category:,link_category:,show_parent_category:,link_parent_category:,show_author:,link_author:,show_create_date:,show_modify_date:,show_publish_date:,show_item_navigation:,show_vote:,show_readmore:,show_readmore_title:,show_hits:,show_noauth:,show_feed_link:1,feed_summary:,menu-anchor_title:,menu-anchor_css:,menu_image:,menu_text:1,page_title:,show_page_heading:1,page_heading:,pageclass_sfx:,menu-meta_description:,menu-meta_keywords:,robots:,secure:0}', 35, 36, 1, '*', 0, '0000-00-00 00:00:00', '0000-00-00 00:00:00');
```
Comment (SharkyKZ) [inline installation/sql/mysql/joomla.sql]: Revert JSON changes.

**Comment by richard67** [comment]
@alikon There is something wrong, see following comparison just for the 2nd record, but it applies to all of them except of the 1st which is special:

Current, i.e. without your PR applied:

```
INSERT INTO `#__menu` (`id`, `menutype`, `title`, `alias`, `note`, `path`, `link`, `type`, `published`, `parent_id`, `level`, `component_id`, `checked_out`, `checked_out_time`, `browserNav`, `access`, `img`, `template_style_id`, `params`, `lft`, `rgt`, `home`, `language`, `client_id`)
SELECT 2, 'main', 'com_banners', 'Banners', '', 'Banners', 'index.php?option=com_banners', 'component', 1, 1, 1, `extension_id`, 0, '0000-00-00 00:00:00', 0, 0, 'class:banners', 0, '', 1, 10, 0, '*', 1 FROM `#__extensions` WHERE `name` = 'com_banners';
```

New, i.e. with your PR applied:

```
INSERT INTO `#__menu` (`id`, `menutype`, `title`, `alias`, `note`, `path`, `link`, `type`, `published`, `parent_id`, `level`, `component_id`, `checked_out`, `checked_out_time`, `browserNav`, `access`, `img`, `template_style_id`, `params`, `lft`, `rgt`, `home`, `language`, `client_id`, `publish_up`, `publish_down`) VALUES
...
(2, 'main', 'com_banners', 'Banners', '', 'Banners', 'index.php?option=com_banners', 'component', 1, 1, 1, 4, 0, '0000-00-00 00:00:00', 0, 0, 'class:banners', 0, '', 9, 18, 0, '*', 1, '0000-00-00 00:00:00', '0000-00-00 00:00:00'),
```

I.e. currently `parent_id` is selected from the `extension_id` column of the `#__extensions` table, but with your PR applied, hard-coded `parent_id` is used like it was in old times.

Or am I reading something wrong when looking at the changes?

**Comment by alikon** [comment]
yes right i've just used export  cause we are on a install script, but  i'll switch back to insert select

**Comment by richard67** [comment]
@alikon Why closed? Will you make a new PR for that? Or has it meanwhile been fixed by someone else?

**Comment by alikon** [comment]
there is  no added value in fixing the install script only... when some strange things may happen during some crazy update script... i'm re-thinking .... after reading  this https://github.com/joomla/joomla-cms/pull/26289#issuecomment-531549468

**Comment by richard67** [comment]
I see. Makes sense. Maybe we should really call a rebuild function (if there is one) after each update or database fix?

**Comment by alikon** [comment]
more i look at the updates script...more i think we are doing it wrong....still trying to understand why we have 3.8.x or 3.9.x   sql update script  when it is supposed to upgrade from 3.10 only....

**Comment by richard67** [comment]
Maybe @wilsonge knows that, why we have update scripts 2.5 to 3.9 in the 4.0-dev branch.

**Comment by wilsonge** [comment]
3.8 can be removed. I'm leaving 3.9 for now because i don't have 3.10-dev merged into 4.0-dev - so i'm testing the 3.x upgrade path from 3.9 for now. Once I've got 3.10-dev in sync with 4.0-dev then we'll remove the 3.9 upgrade files too

We don't have the 2.5-3.7 files @richard67 ;)

**Comment by richard67** [comment]
@wilsonge It seems it's a while ago I've looked on the update sql scripts ;-)

**Pull request closed without merging**